lib/ansible/modules/web_infrastructure/wildfly.py

```diff
@@ -0,0 +1,309 @@
+
+ANSIBLE_METADATA = {'status': ['preview'], 'supported_by': 'community',
+                    'metadata_version': '1.0'}
+
+DOCUMENTATION = \
+    """
+module: wildfly
+version_added: "2.4"
+short_description: deploy applications to Wildfly
+description:
+  - Deploy applications to Wildfly domain and standalone mode
+options:
+  deployment:
+    required: true
+    description:
+      - The name of the deployment
+  src:
+    required: false
+    description:
+      - The remote path of the application ear or war to deploy
+  controller_address:
+    required: true
+    default: http://localhost:9990
+    description:
+      - The http api management address
+  state:
+    required: false
+    choices: [ present, absent ]
+    default: "present"
+    description:
+      - Whether the application should be deployed or undeployed
+  username:
+    required: true
+    description:
+      - The http api management username
+  password:
+    required: true
+    description:
+      - The http api management password
+  mode:
+    required: false
+    default: domain
+    description: Domain mode or standalone mode
+  server_groups:
+    required: false
+    default: all-relevant-server-groups
+    description: Server groups to deploy the war
+notes:
+author: "Aditya C S"
+"""
+
+EXAMPLES = \
+    """
+# Deploy a hello world application
+- wildfly:
+    src: /tmp/hello-1.0-SNAPSHOT.war
+    deployment: hello.war
+    state: present
+    controller_address: http://localhost:9990
+    username: username
+    password: password
+    mode: domain
+    server_groups: all-relevant-server-groups
+# Undeploy the hello world application
+- wildfly:
+    deployment: hello.war
+    state: absent
+    controller_address: http://localhost:9990
+    username: username
+    password: password
+    mode: domain
+    server_groups: all-relevant-server-groups
+"""
+
+RETURN = \
+    '''
+---
+username:
+  description: username used for authentication.
+  returned: success
+  type: string
+  sample: admin
+state:
+  description: State of the deployment.
+  returned: success
+  type: string
+  sample: present
+password:
+  description: password used for authentication.
+  returned: success
+  type: string
+  sample: admin
+controller_address:
+  description: Url to connect to the wilddly management server.
+  returned: success
+  type: string
+  sample: http://localhost:9990
+'''
+
+try:
+    import json
+except ImportError:
+    try:
+        import simplejson as json
+    except ImportError:
+
+        # Let snippet from module_utils/basic.py return a proper error in this
+        # case
+
+        pass
+import ansible.module_utils.six.moves.urllib.error as urllib_error
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import open_url, url_argument_spec
+from ansible.module_utils.pycompat24 import get_exception
+
+
+def undeploy(deployment, server_groups):
+    deployment = '{}'.format(deployment)
```

Comment:
Why are you formatting the variable? The same in many places in the code below. Is the reason that you want to have it as `string`? In that case, you can define `type='str'` in the `argument_spec` to enforce that type.